growth/too/tasks/decam_client.py

```diff
@@ -0,0 +1,85 @@
+
+import sys
+
+from ..flask import app
+from ..models import db
+from astropy import time
+import astropy.units as u
+from astropy import table
+from astropy.coordinates import SkyCoord
+from sqlalchemy import Table, orm
+from celery.task import PeriodicTask
+from celery.utils.log import get_task_logger
+from celery.local import PromiseProxy
+import numpy as np
+import pyvo.dal
+import pkg_resources
+
+from . import celery
+from .. import models
+
+log = get_task_logger(__name__)
+
+
+
+class DECamTiles(db.Model):
+    """ Fetch the DECam tiles from the database."""
+    __bind_key__ = 'decam_db'
+    __table__ = Table('tiles', db.metadata, autoload=True,
+    autoload_with=db.get_engine(app, bind=__bind_key__))
+
+
+@celery.task(base=PeriodicTask, shared=False, run_every=3600)
+def decam_obs(start_time=None, end_time=None):
+    """ Periodically query the DECam database to get the properties of the observed tiles."""
+
+    if start_time is None:
+        start_time = time.Time.now() - time.TimeDelta(1.0*u.day)
+    if end_time is None:
+        end_time = time.Time.now()
+
+    # get all the DECam tiles, filter by mjd_obs
+    query = db.session.query(DECamTiles).filter(DECamTiles.mjd_obs.between(start_time.mjd, end_time.mjd))
+    tiles = query.all()
+
+    bands = {'g': 1, 'r': 2, 'i': 3, 'z': 4, 'J': 5, 'U': 6, 'a': 7}
+    fields = ( 'exptime', 'ra', 'dec', 'field_id', 'id', 'filter_id', 'subfield_id', 'airmass', 'maglim')
+
+    tiles_table = table.Table(names=fields)
+
+    tessfile = pkg_resources.resource_stream(__name__, '../input/DECam.tess')
```

Comment:
Use a `with:` statement here so that you close the file after you are done with it.